python/ray/autoscaler/_private/aws/config.py

```diff
@@ -414,6 +418,36 @@ def _configure_subnet(config):
     return config
 
 
+def _vpc_filter_from_sg(config):
+    """Returns a filter that restricts resources to ones with the same vpc-id
+    as the security groups specified in config's head_node and worker_node
+    fields.
+
+    If neither head nor worker specifies a security group, returns an empty
+    filter.
+    """
+    head_sg_ids = config["head_node"].get("SecurityGroupIds", [])
+    worker_sg_ids = config["worker_nodes"].get("SecurityGroupIds", [])
+    sg_ids = head_sg_ids + worker_sg_ids
+    sg_ids = list(set(sg_ids))
+
+    ec2 = _resource("ec2", config)
+    filters = [{"Name": "group-id", "Values": sg_ids}]
+    security_groups = ec2.security_groups.filter(Filters=filters)
+    vpc_ids = [sg.vpc_id for sg in security_groups]
+    vpc_ids = list(set(vpc_ids))
+
+    multiple_vpc_msg = "All security groups specified in the cluster config "\
+        "should belong to the same VPC."
+    cli_logger.doassert(len(vpc_ids) <= 1, multiple_vpc_msg)
+    assert len(vpc_ids) <= 1, multiple_vpc_msg
+
```

Comment:
Note: 
If the user specifies security groups for both head and worker, and the security groups belong to different VPCs, this will throw an error. 

We could have it support user-specified security groups in different VPCs for head and worker.
This would match the behavior for subnets: https://github.com/ray-project/ray/pull/8374 .
I think I won't bother with that, unless reviewers think I should.

Reply:
If we wanted to go further down that path, we would want at some point to support different VPCs for each node type in a multi node-type config.